.github/workflows/cherry-pick.yml

```diff
@@ -0,0 +1,151 @@
+name: Cherry Pick
+
+on:
+  issue_comment:
+    types: [created]
+  pull_request_target:
+    types: [closed]
+
+jobs:
+  cherry-pick:
+    # This job will run when a PR is merged with a specific comment,
+    # or when a comment is added to an already merged PR.
+    runs-on: ubuntu-latest
+    # Use pull_request_target so that we can add comments back to the PR
+    # if the cherry-pick fails.
+    permissions:
+      pull-requests: write
+      contents: write
+      issues: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Required to get all history for cherry-picking
+
+      - name: Extract info and determine trigger
+        id: info
+        run: |
+          # Default to failure unless a valid trigger is found
+          echo "valid=false" >> $GITHUB_OUTPUT
+
+          if [[ "${{ github.event_name }}" == "pull_request_target" && "${{ github.event.pull_request.merged }}" == "true" ]]; then
+            echo "Triggered by PR merge"
+            PR_NUMBER="${{ github.event.pull_request.number }}"
+
+            # Check PR body first, then fall back to comments
+            TEXT_TO_SEARCH="${{ github.event.pull_request.body }}"
+            if [[ ! "$TEXT_TO_SEARCH" =~ /cherry-pick[[:space:]]+(stable|preview) ]]; then
+              echo "Command not found in PR body. Checking comments..."
+              TEXT_TO_SEARCH=$(gh pr view $PR_NUMBER --json comments -q '.comments[].body' | tail -n 100)
+            fi
+
+            if [[ "$TEXT_TO_SEARCH" =~ /cherry-pick[[:space:]]+(stable|preview) ]]; then
```

Comment:
This is confusing to do have the same regex in many places. The first two times are POSIX regex and the third is a PCRE capturing.

This should probably be a single PCRE regex and not be duplicated. It should also like include. ^ anchor at the beginning.